### PR TITLE
chor(apollo): Add note about incompatibility with Apollo4

### PR DESCRIPTION
**Pull request by kahest**
## DESCRIBE YOUR PR

Adds notes about incompatibility with Apollo4 in our Android+Java pages.

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [x] Other deadline: ~next week (no urgent deadline, but Apollo4 is released so the information is already relevant)
- [ ] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [x] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## EXTRA RESOURCES

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: docs/platforms/android/integrations/apollo/index.mdx [hunk 1]
```diff
@@ -16,6 +16,12 @@ Capturing transactions requires that you first <PlatformLink to="/tracing/">set
 
 Sentry Apollo integration provides the `SentryApolloInterceptor`, which creates a span for each outgoing HTTP request executed with an [Apollo Android](https://github.com/apollographql/apollo-android) GraphQL client.
 
+<Note>
+
+This integration does not yet support <Link to="https://github.com/apollographql/apollo-kotlin/releases/tag/v4.0.0">Apollo4</Link>. If you're using Apollo4 and are interested in an integration for it, please track and upvote the <Link to="https://github.com/getsentry/sentry-java/issues/3662">related issue</Link>.
```
Comment (adinauer) [inline docs/platforms/android/integrations/apollo/index.mdx:21]: Should we rephrase this to say it's only intended for Apollo v2? Same below for v3. My guess is, we'll have to build another module for Apollo v4.

**Review by adinauer — APPROVED** [review]
See comment

File: docs/platforms/android/integrations/apollo/index.mdx (hunk 1 above)
Comment (kahest) [reply docs/platforms/android/integrations/apollo/index.mdx]: @adinauer updated wording, ptal 🙏

**Review by vivianyentran — APPROVED** [review]

**Review by adinauer — APPROVED** [review]
LGTM!
